### PR TITLE
Bm60rgb iso nordic keymap with sleepmode functional

### PR DESCRIPTION
**Pull request by JakobST1n**
## Description
This is an additional keymap for the bm60rgb_iso board.

- Better suited for nordic layouts
  - `KC_GRAVE` on second layer instead of having `KC_GESC` as that is problematic with nordic layout and macos. (grave giving `'`)
  - Button  right of `LSHIFT` is `KC_NUBS` as that follows the "norm".
- Some general modifications to layout
  - Right-shift has mod-tap with `KC_SLSH` as that button has a bit weird location on this board.
  - On the second layer the arrow cluster is media keys.
  - HJKL is home/end pg up/down on second layer (Nice for anyone used to vim).
  - Backspace is delete on second
  - numbers are function row.
- Also added a sleepmode function. This, separately from usb suspend and such, sets the rgb_matrix effects to a "sleep mode" when no keypresses have been detected for 10 minutes (default sleep effect is "off").

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/config.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*****************************
```
Comment (drashna) [inline keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/config.h:17]:
```suggestion
#pragma once

/*****************************
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/keymap.c [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+         _______, _______, _______, _______, _______, _______, KC_HOME, KC_PGDN, KC_PGUP,  KC_END, _______, _______, _______, _______,
+         _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_VOLD, KC_VOLU,
+         _______, _______, _______,                      _______,                         _______, _______, KC_MEDIA_PREV_TRACK, KC_MEDIA_PLAY_PAUSE, KC_MEDIA_NEXT_TRACK
+    ),
+};
+
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    #ifdef RGB_MATRIX_ENABLE
+    #ifdef SLEEPMODE_ENABLE
```
Comment (drashna) [inline keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/keymap.c:73]: This is a bit less verbose, easier to read, and formatted correctly. 
```suggestion
#if defined(RGB_MATRIX_ENABLE) && defined(SLEEPMODE_ENABLE)
```

File: keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/keymap.c [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                rgb_matrix_mode_noeeprom(sleepmode_before_mode);
+                rgb_matrix_set_speed_noeeprom(sleepmode_before_anim_speed);
+                rgb_matrix_sethsv_noeeprom(rgb_matrix_get_hue(), rgb_matrix_get_sat(), sleepmode_before_brightness);
+                sleepmode_on = false;
+            }
+            idle_timer = timer_read();
+            halfmin_counter = 0;
+        }
+    #endif
+    #endif
```
Comment (drashna) [inline keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/keymap.c:90]:
```suggestion
#endif
```

File: keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/keymap.c [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    return true;
+ }
+
+ void matrix_scan_user(void) {
+     #ifdef RGB_MATRIX_ENABLE
+     #ifdef SLEEPMODE_ENABLE
```
Comment (drashna) [inline keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/keymap.c:96]:
```suggestion
#if defined(RGB_MATRIX_ENABLE) && defined(SLEEPMODE_ENABLE)
```

File: keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/keymap.c [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            //rgb_matrix_disable_noeeprom();
+
+            rgb_matrix_mode_noeeprom(SLEEPMODE_RGB_MODE);
+            rgb_matrix_set_speed_noeeprom(SLEEPMODE_RGB_ANIMATION_SPEED);
+            rgb_matrix_sethsv_noeeprom(rgb_matrix_get_hue(), rgb_matrix_get_sat(), SLEEPMODE_RGB_VAL);
+            sleepmode_on = true;
+            halfmin_counter = 0;
+        }
+    #endif
+    #endif
```
Comment (drashna) [inline keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/keymap.c:119]:
```suggestion
#endif
```

**Review by drashna — COMMENTED** [review]
the sleepmode appears to be the same or similar to `RGB_DISABLE_TIMEOUT` in the core feature.

https://docs.qmk.fm/#/feature_rgb_matrix?id=additional-configh-options

**Comment by JakobST1n** [comment]
> the sleepmode appears to be the same or similar to `RGB_DISABLE_TIMEOUT` in the core feature.
> 
> https://docs.qmk.fm/#/feature_rgb_matrix?id=additional-configh-options

It is kind of similar, but the added functionality of setting any animation mode on timeout. Rather that just off.
Might be better to add this functionality to core?
Although then I would have to read up on timers, as at least on my board I believe most of them are already in in use...

**Comment by drashna** [comment]
Ah, okay.  

And no, there isn't.  But... maybe there should be, because I have something similar implemented in my userspace. :D

**Review by drashna — APPROVED** [review]
It looks like this needs a formatting pass, but otherwise, looks good.

File: keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/keymap.c [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+     * |    |   |   |   |   |   |   |   |   |   |   |      |V_D|V_U|
+     * |-----------------------------------------------------------|
+     * |    |    |    |                            |   |PRE|PLA|NEX|
+     * `-----------------------------------------------------------' */
+    [1] = LAYOUT(
+        KC_GRAVE,   KC_F1,   KC_F2,     KC_F3, KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,  KC_F12, KC_DEL,
+         _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______, _______,
+         _______, _______, _______, _______, _______, _______, KC_HOME, KC_PGDN, KC_PGUP,  KC_END, _______, _______, _______, _______,
+         _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_VOLD, KC_VOLU,
+         _______, _______, _______,                      _______,                         _______, _______, KC_MEDIA_PREV_TRACK, KC_MEDIA_PLAY_PAUSE, KC_MEDIA_NEXT_TRACK
```
Comment (noroadsleft) [inline keyboards/bm60rgb_iso/keymaps/iso_nordic_sleepmode/keymap.c:66]: No logic change; just makes this more readable.

```suggestion
         KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
         _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______, _______,
         _______, _______, _______, _______, _______, _______, KC_HOME, KC_PGDN, KC_PGUP, KC_END,  _______, _______, _______, _______,
         _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_VOLD, KC_VOLU,
         _______, _______, _______,                      _______,                         _______, _______, KC_MPRV, KC_MPLY, KC_MNXT
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**